taskslist/app/controllers/tasks_controller.rb

```diff
@@ -7,4 +7,36 @@ def show
     id = params[:id]
     @task = Task.find(id)
   end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    Task.create(task_params[:task])
+    redirect_to '/tasks/'
+  end
+
+  def destroy
+    Task.destroy(params["id"])
+    redirect_to '/tasks/'
+  end
+
+  def complete
+    task = Task.find(params[:id])
+    task.done_date = Time.now
+    task.save
+    redirect_to '/tasks/'
+  end
+
+  def edit
+    @task = Task.find(params[:id])
+    @edit = true
+    render 'tasks/new'
```

Comment:
If you're going to re-use the new view for the edit, then I would recommend renaming it so that it isn't new